tests/mpi/collective_min_vector_in_place.cc

```diff
@@ -0,0 +1,67 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2009 - 2015 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+
+// check Utilities::MPI::min for vectors, but with input=output
+
+#include "../tests.h"
+#include <deal.II/base/logstream.h>
+#include <deal.II/base/utilities.h>
+#include <fstream>
+
+void test()
+{
+  unsigned int myid = Utilities::MPI::this_mpi_process (MPI_COMM_WORLD);
+  const unsigned int numprocs = Utilities::MPI::n_mpi_processes (MPI_COMM_WORLD);
+
+  unsigned int values_[2] = { 1, 2 };
+  std::vector<unsigned int> inout(&values_[0], &values_[2]);
```

Comment:
It might make sense to set the entries to `1+myid` and `2+myid`, respectively, to really test that we do perform a minimization (I believe your code and it is implicitly tested by the generic implementation and the scalar case, but just to be sure). The same for the array version above.